substrate/client/db/src/lib.rs

```diff
@@ -804,9 +804,16 @@ impl<Block: BlockT> sc_client_api::blockchain::Backend<Block> for BlockchainDb<B
 		}
 	}
 
-	fn clear_block_gap(&self) {
+	fn clear_block_gap(&self) -> ClientResult<()> {
 		debug!(target: "sync", "Clear block gap.");
+
+		let mut transaction = Transaction::new();
+		transaction.remove(columns::META, meta_keys::BLOCK_GAP);
```

Comment:
Do you know if there are any side effects in the blockchain besides this gap that we need to be prepared for?

Reply:
I know only about common blocks with peers in sync engine cache. It is addressed in the `restart_sync` method along with Full sync mode toggle. It's not critical and auto-recovers but produces error messages sometimes if not addressed with `restart`.